.github/workflows/test_pull_requests.yml

```diff
@@ -0,0 +1,199 @@
+      working-directory: ./baseline/build
+      run: >
+        cmake 
+        -G "${{ matrix.generator }}"
+        -DCMAKE_BUILD_TYPE:STRING=Release
+        -DCMAKE_OSX_DEPLOYMENT_TARGET:STRING=${{ matrix.macos_dev_target }}
+        -DLINK_WITH_PYTHON:BOOL=ON
+        -DPython_REQUIRED_VERSION:STRING=${{ steps.setup-python.outputs.python-version }}
+        -DPython_ROOT_DIR:PATH=$RUNNER_TOOL_CACHE/Python/${{ steps.setup-python.outputs.python-version }}/${{ matrix.python-arch }}/
+        -DBUILD_TESTING:BOOL=ON
+        -DBUILD_FORTRAN:BOOL=ON 
+        -DBUILD_PACKAGE:BOOL=OFF
+        -DDOCUMENTATION_BUILD:STRING=DoNotBuild
+        -DENABLE_OPENMP:BOOL=OFF
+        -DUSE_OpenMP:BOOL=OFF
+        ../
+
+    # During baseline builds, just build specific target list so that we don't waste time building the unit test binary
+
+    - name: Baseline Build on Windows
+      if: matrix.run_regressions && runner.os == 'Windows'
+      working-directory: ./baseline/build
+      run: cmake --build . -j ${{ matrix.nproc }} --config Release --target energyplus ExpandObjects_build ReadVars_build Slab_build Basement_build AppGPostProcess_build ParametricPreprocessor_build
+
+    - name: Baseline Build on Mac/Linux
+      if: matrix.run_regressions && runner.os != 'Windows'
+      working-directory: ./baseline/build
+      run: cmake --build . -j ${{ matrix.nproc }} --target energyplus ExpandObjects ReadVarsESO Slab Basement AppGPostProcess ParametricPreprocessor
+
+    - name: Baseline Test
+      if: matrix.run_regressions
+      working-directory: ./baseline/build
+      run: ctest -C Release -E Basement -R integration -j 3  # TODO: Speed up basement so we don't have to skip it.
+
+    # BUILD AND TEST EVERYTHING ON THE CURRENT BRANCH
+
+    - name: Branch Checkout
+      uses: actions/checkout@v4
+      with:
+        path: branch
+
+    - name: Branch Create Build Directory
+      run: cmake -E make_directory ./branch/build/
+
+    - name: Branch Configure CMake
+      working-directory: ./branch/build
+      run: >
+        cmake 
+        -G "${{ matrix.generator }}"
+        -DCMAKE_BUILD_TYPE:STRING=Release
+        -DCMAKE_OSX_DEPLOYMENT_TARGET:STRING=${{ matrix.macos_dev_target }}
+        -DLINK_WITH_PYTHON:BOOL=ON
+        -DPython_REQUIRED_VERSION:STRING=${{ steps.setup-python.outputs.python-version }}
+        -DPython_ROOT_DIR:PATH=$RUNNER_TOOL_CACHE/Python/${{ steps.setup-python.outputs.python-version }}/${{ matrix.python-arch }}/
+        -DBUILD_TESTING:BOOL=ON
+        -DBUILD_FORTRAN:BOOL=ON
+        -DBUILD_PACKAGE:BOOL=OFF
+        -DDOCUMENTATION_BUILD:STRING=DoNotBuild
+        -DENABLE_OPENMP:BOOL=OFF
+        -DUSE_OpenMP:BOOL=OFF
+        ../
+
+    - name: Branch Build
+      id: branch_build
+      working-directory: ./branch/build
+      run: cmake --build . -j ${{ matrix.nproc }} --config Release
+
+    - name: Branch Test
+      working-directory: ./branch/build
+      run: ctest -C Release -E Basement -j 3
```

Comment:
Yeah, it would be great if there was a script already available that could process the ctest results.  If there is, I'm open to bringing it in and using it.  But developing that would probably take a good several hours from an experienced dev.